03_Modules/SmartCharging/src/module/api.ts

```diff
@@ -91,12 +100,322 @@ export class SmartChargingModuleApi
+        };
+      }
+
+      if (
+        chargingProfile.chargingProfileKind ===
+          ChargingProfileKindEnumType.Absolute ||
+        chargingProfile.chargingProfileKind ===
+          ChargingProfileKindEnumType.Recurring
+      ) {
+        // OCPP 2.0.1 Part 2 K01.FR.40
+        if (!chargingSchedule.startSchedule) {
+          return {
+            success: false,
+            payload: `ChargingSchedule ${chargingSchedule.id}: startSchedule SHALL be set when chargingProfileKind is Absolute or Recurring.`,
+          };
+        }
+      } else if (
+        chargingProfile.chargingProfileKind ===
+        ChargingProfileKindEnumType.Relative
+      ) {
+        // OCPP 2.0.1 Part 2 K01.FR.41
+        if (chargingSchedule.startSchedule) {
+          return {
+            success: false,
+            payload: `ChargingSchedule ${chargingSchedule.id}: startSchedule SHALL be absent when chargingProfileKind is Relative.`,
+          };
+        }
+      }
+
+      // OCPP 2.0.1 Part 2 K01.FR.26
+      if (
+        chargingScheduleChargingRateUnit &&
+        chargingScheduleChargingRateUnit.dataType === DataEnumType.MemberList &&
+        chargingScheduleChargingRateUnit.valuesList
+      ) {
+        try {
+          const chargingRateUnits = stringToSet(
+            chargingScheduleChargingRateUnit.valuesList,
+          );
+          if (!chargingRateUnits.has(chargingSchedule.chargingRateUnit)) {
+            return {
+              success: false,
+              payload: `ChargingSchedule ${chargingSchedule.id}: chargingRateUnit SHALL be one of ${chargingScheduleChargingRateUnit.valuesList}.`,
+            };
+          }
+        } catch (error) {
+          this._logger.error(
+            `Failed to validate chargingRateUnit. Found unexpected valueList in RateUnit: ${JSON.stringify(chargingScheduleChargingRateUnit.valuesList)}`,
+            error,
+          );
+        }
+      }
+
+      // OCPP 2.0.1 Part 2 K01.FR.35
+      chargingSchedule.chargingSchedulePeriod.sort((period1, period2) => {
+        if (period1.startPeriod > period2.startPeriod) {
+          return 1;
+        }
+        if (period1.startPeriod < period2.startPeriod) {
+          return -1;
+        }
+        return 0;
+      });
+
+      for (const chargingSchedulePeriod of chargingSchedule.chargingSchedulePeriod) {
+        if (chargingSchedulePeriod.phaseToUse) {
+          // OCPP 2.0.1 Part 2 K01.FR.19
+          if (chargingSchedulePeriod.numberPhases !== 1) {
+            throw new Error(
+              `ChargingSchedule ${chargingSchedule.id}: PhaseToUse SHALL only be set with numberPhases = 1.`,
+            );
+          }
+          // OCPP 2.0.1 Part 2 K01.FR.20
+          if (
+            acPhaseSwitchingSupported.length === 0 ||
+            !acPhaseSwitchingSupported[0].value
+          ) {
+            throw new Error(
```

Comment:
This doesn't look like it's gets caught anywhere, like the ones in the validator. So this won't get formatted into the response format. 
Probably should just return in the correct format.